Various fixes

jakirkham opened the pull request:
* Fix dependencies for 1.0.
* Build and install with `pip` (per [the recommendation]( https://conda-forge.org/docs/meta.html#use-pip )).
* ~~Use the `sdist` from PyPI.~~ ( blocked by PR https://github.com/lanpa/tensorboard-pytorch/pull/125 )

jakirkham committed:
Change `fn` to include `name` and `version`

Should make it a little easier to find the tarball in `conda-build`'s
source cache if needed.

jakirkham committed:
Fix up dependencies

* `protobuf` has a lower bound in terms of an acceptable version.
* `six` is also a dependency of the current version.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
Build and install with `pip`

As the recommended build practice in `conda-forge` is to build and
install with `pip` (particularly for `noarch: python` packages), switch
over to using `pip` in the `script` step.

jakirkham committed:
Bump build number to 1

Rebuild the package as there have been dependency corrections and `pip`
is now used to do the build.
